Sources/MuxPlayerSwift/FairPlay/FairPlaySessionManager.swift

```diff
@@ -187,38 +193,53 @@ class DefaultFairPlayStreamingSessionManager<
                 self.logger.debug(
                     "Applicate certificate request failed with error: \(error.localizedDescription)"
                 )
+                let error = FairPlaySessionError.because(cause: error)
                 requestCompletion(Result.failure(
-                    FairPlaySessionError.because(cause: error)
+                    error
                 ))
+                self.errorDispatcher.dispatchApplicationCertificateRequestError(
+                    error: error,
+                    playbackID: playbackID
+                )
                 return
             }
             // error case: I/O finished with non-successful response
             guard responseCode == 200 else {
                 self.logger.debug(
                     "Applicate certificate request failed with response code: \(String(describing: responseCode))"
                 )
+                let error = FairPlaySessionError.httpFailed(
+                    responseStatusCode: responseCode ?? 0
+                )
                 requestCompletion(
                     Result.failure(
-                        FairPlaySessionError.httpFailed(
-                            responseStatusCode: responseCode ?? 0
-                        )
+                        error
                     )
                 )
+                self.errorDispatcher.dispatchApplicationCertificateRequestError(
+                    error: error,
+                    playbackID: playbackID
+                )
                 return
             }
             // this edge case (200 with invalid data) is possible from our DRM vendor
             guard let data = data,
                   data.count > 0 else {
+                let error = FairPlaySessionError.unexpected(
+                    message: "No cert data with 200 OK respone"
```

Comment:
nitpick(blocking): looks like a preexisting condition, but typo. ("respone" instead of "response")

Reply:
Good catch

Reply:
9116546bfc85cd537bae1f7334588b703facde85